src/service/authentication-service.ts

```diff
@@ -268,9 +268,10 @@ export default class AuthenticationService extends WithManager {
       const { searchEntries } = await client.search(ldapSettings.base, {
         scope: 'sub',
         filter: filterstr,
+        explicitBufferAttributes: ['objectGUID'],
       });
       if (searchEntries[0]) {
-        ADUser = userFromLDAP(searchEntries[0]);
+        ADUser = userFromLDAP(searchEntries[0] as any as LDAPResult);
```

Comment:
Same here.

Reply:
This is directly from the `client.search`, so no.

Reply:
Then why the need to cast it, if `searchEntries[0]` is already of type `any`?

Reply:
It's not, its of type Entry which is an object of key-value pairs with strings as key. They decided not to specify any of the keys, not even the default AD ones which we use.

Reply:
A possible refactor would be to have a helper function which turns the Entry into an LDAPResult and even type checks it for sanity, however I do think this is out-of-scope of this PR. Previously before this PR the any to type conversion just happened in the `userFromLDAP` function itself, so nothing changed.